django_project/changes/tests/test_views.py

```diff
@@ -752,6 +752,349 @@ def test_VersionDelete_no_login(self):
+    """
+    Check if normal(a regular logged in user) user can perform CRUD
+    operations on version entries just in case they have the URL to the views.
+    """
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def setUp(self):
+        """
+        Setup before each test
+        We force the locale to en otherwise it will use
+        the locale of the host running the tests and we
+        will get unpredictable results / 404s
+        """
+
+        self.client = Client()
+        self.client.post(
+            '/set_language/', data={'language': 'en'})
+        logging.disable(logging.CRITICAL)
+        self.project = ProjectF.create(name='testproject')
+        self.version = VersionF.create(
+            project=self.project,
+            name='1.0.1')
+        self.category = CategoryF.create(
+            project=self.project,
+            name='testcategory')
+
+        self.user = UserF.create(**{
+            'username': 'sonlinux',
+            'password': 'password',
+        })
+        # Something changed in the way factoryboy works with django 1.8
+        # I think - we need to explicitly set the users password
+        # because the core.model_factories.UserF._prepare method
+        # which sets the password is never called. Next two lines are
+        # a work around for that - sett #581
+        self.user.set_password('password')
+        self.user.save()
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def tearDown(self):
+        """
+        Teardown after each test.
+
+        :return:
+        """
+        self.project.delete()
+        self.version.delete()
+        self.category.delete()
+        self.user.delete()
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionListView_with_normal_user(self):
+        """
+        Test if a normal user can view a list of version entries.
+        """
+        self.client.login(username='sonlinux', password='password')
+        response = self.client.get(reverse('version-list', kwargs={
+            'project_slug': self.project.slug
+        }))
+
+        expected_templates = [
+            'version/list.html', u'changes/version_list.html'
+        ]
+
+        self.assertEqual(
+            response.status_code, 200)
+        self.assertEqual(
+            response.template_name, expected_templates)
+        self.assertEqual(
+            response.context_data['object_list'][0],
+                         self.version)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionCreateView_with_normal_user(self):
+        """
+        Test if a normal user can create a list of version entries.
+        """
+        self.client.login(username='sonlinux', password='password')
+        response = self.client.get(reverse('version-create', kwargs={
+            'project_slug': self.project.slug
+        }))
+
+        self.assertEqual(
+            response.status_code, 200)
```

Comment:
This is wrong I guess. A normal user can't add a new version. (only changelog manager or staff)

Can you update the view and also the HTML template to check that the button is not visible.

Reply:
Thanks @Gustry 
Let me do so.

Reply:
So a normal user can add a new version only if he is the project owner. (or any staff or changelog manager)

Reply:
Fit on one line